Assets/Scripts/New Scripts/InputSystem.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class InputSystem : MonoBehaviour
+{
+    public Vector2 InputDir { get; private set; }
+
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        InputDir = Vector2.zero;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        InputDir = CalculateMoveInput();
+    }
+
+    private Vector2 CalculateMoveInput()
+    {
+        // Movement Input
+        bool moveUp = Input.GetKey(KeyCode.UpArrow) || Input.GetKey(KeyCode.W);
+        bool moveDown = Input.GetKey(KeyCode.DownArrow) || Input.GetKey(KeyCode.S);
+        bool moveLeft = Input.GetKey(KeyCode.LeftArrow) || Input.GetKey(KeyCode.A);
+        bool moveRight = Input.GetKey(KeyCode.RightArrow) || Input.GetKey(KeyCode.D);
+
+        Vector2 move = Vector2.zero;
+
+        if (moveUp)
+        {
+            move.y += 1;
+        }
+
+        if (moveDown)
+        {
+            move.y -= 1;
+        }
+
+        if (moveLeft)
+        {
+            move.x -= 1;
+        }
+
+        if (moveRight)
+        {
+            move.x += 1;
+        }
```

Comment:
this code could be simplified using the following idea... 

```
var vertical = Input.GetAxisRaw("Vertical");
var horizontal = Input.GetAxisRaw("Horrizontal");

move = new Vector2(vertical, horizontal).normalized;
```

if you haven't done anything with the input manager, the get axis should by default support directional keys and wasd. 

`GetAxisRaw` will give you no interpolation for the axis you request. (i.e. will always return 0, 1, or -1)

Reply:
I have changed it to the simplified version

https://github.com/jamesantiporda/Encore/blob/3c3b5fb67694e6dd1ba0546d3f8f93e8d63d1028/Assets/Scripts/New%20Scripts/InputSystem.cs#L22-L31